### PR TITLE
Cleanup Structs and Natspec

### PR DESCRIPTION
**Pull request by amiecorso**
Moved the structs out, small natspec cleanups

### CODE REVIEW

File: src/SpendPermissionManager.sol [hunk 1]
```diff
@@ -301,14 +303,11 @@ contract SpendPermissionManager is EIP712 {
...
-    ///
-    /// @param spendPermissionBatch Details of the spend permission batch.
-    /// @param signature Signed approval from the user.
-    ///
-    /// @return allApproved True if all spend permissions in the batch are approved and not revoked.
+    /// @dev Compatible with ERC-6492 signatures. Does not enforce uniqueness of permissions 
+    ///      within a batch, allowing duplicate idempotent approvals.
+    /// @param spendPermissionBatch Details of the spend permission batch
+    /// @param signature Signed approval from the user
+    /// @return allApproved True if all spend permissions in the batch are approved and not revoked
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:310]: https://github.com/coinbase/solidity-style-guide?tab=readme-ov-file#3-newlines-between-tag-types
think we want to keep the new-lines

File: src/SpendPermissionManager.sol [hunk 2]
```diff
@@ -722,10 +718,10 @@ contract SpendPermissionManager is EIP712 {
     /// @dev Enforces successful transfer of native token, reverts on failure.
     /// @dev Uses `safeTransferFrom` for ERC-20 token transfers to enforce revert on failure.
     ///
-    /// @param token Address of the token contract.
-    /// @param account Address of the user account.
-    /// @param recipient Address of the token recipient.
-    /// @param value Amount of tokens to transfer.
+    /// @param token Address of token (use NATIVE_TOKEN for ETH)
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:721]:
```suggestion
    /// @param token Address of token (ERC-20 contract or ERC-7528 native token address)
```

**Review by ilikesymmetry — APPROVED** [review]
final gut check from you: does it feel cleaner having the structs out after putting it to paper?

File: src/SpendPermissionManager.sol (hunk 1 above)
Comment (amiecorso) [reply src/SpendPermissionManager.sol:311]: good catch, just gave this a once-over for consistency across all functions again

**Comment by amiecorso** [comment]
yeah, I prefer the structs moved out quite a bit

**Pull request merged**